lib/internal/source_map/source_map_cache.js

```diff
@@ -351,11 +352,53 @@ function findSourceMap(sourceURL) {
   return sourceMap;
 }
 
+/**
+ *
+ * @param {Record<string, unknown>} callSite // The call site object to reconstruct from source map
+ * @returns {Record<string, unknown> | undefined} // The reconstructed call site object
+ */
+function reconstructCallSite(callSite) {
+  const { scriptName, lineNumber, column } = callSite;
+  const sourceMap = findSourceMap(scriptName);
+  if (!sourceMap) return;
+  const entry = sourceMap.findEntry(lineNumber - 1, column - 1);
+  if (!entry?.originalSource) return;
+  return {
+    __proto__: null,
+    functionName: callSite.name,
+    lineNumber: entry.originalLine + 1,
+    column: entry.originalColumn + 1,
+    scriptName: entry.originalSource,
+  };
+}
+
+/**
+ *
+ * The call site object or array of object to map (ex `util.getCallSite()`)
+ * @param {Record<string, unknown> | Record<string, unknown>[]} callSites
+ * An object or array of objects with the reconstructed call site
+ * @returns {Record<string, unknown> | Record<string, unknown>[]}
+ */
+function mapCallSite(callSites) {
+  if (ArrayIsArray(callSites)) {
+    const result = [];
+    for (const callSite of callSites) {
```

Comment:
I think we avoid `const...of` in lib in general because iterators can be slow/unreliable.

```suggestion
    for (let i = 0; i < callSites.length; ++i) {
      const callSite = callSites[i];
```